### PR TITLE
Protection Policy DataSource

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description
Implemented read operation for protection policy datasource and added acceptance tests for it.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Acceptance tests
image

Note: 
"virtual_volumes" and "nas_servers" attributes are not added as these are added in PowerStore version 3.0 and respective changes in gopowerstore might break backward compatibility hence, approach to include these attributes is still to be finalized. It can be tracked through https://jira.cec.lab.emc.com/browse/TFM-448

### CODE REVIEW

File: powerstore/datasource_protection_policy.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+	return &protectionPolicyDataSource{}
+}
+
+func (d *protectionPolicyDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_protectionpolicy"
+}
+
+func (d *protectionPolicyDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: ".",
```
Comment (shenda1) [inline powerstore/datasource_protection_policy.go]: Add the description

File: powerstore/datasource_protection_policy.go [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+			},
+
+			"policies": schema.ListNestedAttribute{
+				Description:         "List of policies.",
+				MarkdownDescription: "List of policies.",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description:         "Unique identifier of the policy.",
```
Comment (shenda1) [inline powerstore/datasource_protection_policy.go]: better to specify protection policy explicitly rather than policy

File: powerstore/datasource_protection_policy_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package powerstore
+
```
Comment (shenda1) [inline powerstore/datasource_protection_policy_test.go:2]: Please add negative scenarios as well

**Comment by AnikaAgiwal2711** [comment]
Can you add the screenshot of the coverage report?

File: powerstore/datasource_protection_policy_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+				Config:      ProtectionPolicyDataSourceparamsNameNegative,
+				ExpectError: regexp.MustCompile("nable to Read PowerStore Protection"),
+			},
+		},
+	})
+}
+
+var ProtectionPolicyDataSourceparamsName = `
+provider "powerstore" {
+	username = "` + username + `"
```
Comment (AnikaAgiwal2711) [inline powerstore/datasource_protection_policy_test.go]: This piece of code is repetitive. Is there a way to avoid it ??

File: powerstore/datasource_protection_policy_test.go (hunk 3 above)
Comment (AnikaAgiwal2711) [reply powerstore/datasource_protection_policy_test.go:2]: Also you can add the test cases with empty id or empty name

File: powerstore/datasource_protection_policy.go (hunk 2 above)
Comment (prajwalpatil25) [reply powerstore/datasource_protection_policy.go]: done

File: powerstore/datasource_protection_policy_test.go (hunk 3 above)
Comment (prajwalpatil25) [reply powerstore/datasource_protection_policy_test.go:2]: done

File: powerstore/datasource_protection_policy_test.go (hunk 4 above)
Comment (prajwalpatil25) [reply powerstore/datasource_protection_policy_test.go]: done

**Comment by AnikaAgiwal2711** [comment]
LGTM. Can I see the coverage once and then I can approve.

**Comment by prajwalpatil25** [comment]
> LGTM. Can I see the coverage once and then I can approve.

@AnikaAgiwal2711 
Added datasource_protection_policy.go coverage snapshot in PR description.

**Review by shenda1 — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]
